umbraco/Umbraco-CMS
# Netcore: Fixes issue with notifications that did not use the SMTP from config

### Notes
- Fixes issue with sending notifications due to the `SMTPClient` not using info from config. Now we use the `IEmailSender`

### Potential improvements
- The `IEmailSender` sends emails async, but the background thread is not async. Right now we use `.GetAwaiter().GetResult();`, maybe this can be done in a better way?

### Test
1. You we need to have STMP set in config
```
{
    "Umbraco": {
        "CMS": {
            "Global": {
                "Smtp": {
                    "From" :"noreply@umbraco.com",
                    "Host": "localhost",
                    "Post": "25"
                }
            }
        }
    }
}
```
2. When smtp in configured, Notifications is available as options on each content. Turn on notifications e.g. for Update
3. Execute the handling that should fire the notification e.g. update and save the content.
4. Verify the email is send. E.g. using [Papercut](https://github.com/ChangemakerStudios/Papercut-SMTP)
